workflow/rules/common.smk

```diff
@@ -826,39 +826,35 @@ def get_depth_input(wildcards):
             illumina_pattern="results/{date}/read-sorted/pe~position/{sample}.hardclipped.bam",
             ont_pattern=expand(
                 "results/{{date}}/mapped/ref~{ref}/{{sample}}.bam",
-                ref=config["adapters"]["amplicon-reference"],
+                ref=config["preprocessing"]["amplicon-reference"],
             ),
             ion_torrent_pattern="results/{date}/read-sorted/se~position/{sample}.hardclipped.bam",
         )
 
     # use trimmed reads
     return "results/{{date}}/mapped/ref~{ref}/{{sample}}.bam".format(
-        ref=config["adapters"]["amplicon-reference"]
+        ref=config["preprocessing"]["amplicon-reference"]
     )
 
 
 def get_adapters(wildcards):
-    # TODO Think about adapter handling. Related #356
-    # Remove the temporary patters later
-    if is_amplicon_data(wildcards.sample):
-        patterns = get_pattern_by_technology(
-            wildcards,
-            illumina_pattern=config["adapters"]["illumina-amplicon"],
-            ont_pattern=config["adapters"]["illumina-amplicon"],
-            ion_torrent_pattern=config["adapters"]["illumina-amplicon"],
-        )
-    else:
-        patterns = get_pattern_by_technology(
-            wildcards,
-            illumina_pattern=config["adapters"]["illumina-shotgun"],
-            ont_pattern=config["adapters"]["illumina-shotgun"],
-            ion_torrent_pattern=config["adapters"]["illumina-shotgun"],
-        )
-
-    if patterns is not None:
-        return patterns
-
-    raise NotImplementedError(f"No adapters implemented for {wildcards.sample}.")
+    try:
+        samples = pep.sample_table
+        samples.dropna(subset=["adapters"], inplace=True)
+        adapters = samples.loc[wildcards.sample]["adapters"]
+
+        # predefined adapters
+        # https://lifesciences.tecan.com/revelo-rna-seq-library-prep-kit
+        if adapters == "revelo-rna-seq":
+            return "--adapter_sequence AGATCGGAAGAGCACACGTCTGAACTCCAGTCA --adapter_sequence_r2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT"
+        # https://www.nimagen.com/shop/products/rc-cov096/easyseq-sars-cov-2-novel-coronavirus-whole-genome-sequencing-kit
+        elif adapters == "nimagen-easy-seq":
+            return "--adapter_sequence GCGAATTTCGACGATCGTTGCATTAACTCGCGAA --adapter_sequence_r2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT"
```

Comment:
I'd still prefer to keep these in the config and not hard coded in the common.smk. Or am I missing something here?

Reply:
These hardcoded sequences are just used if you enter "revelo-rna-seq" or "nimagen-easy-seq" in the new adapter column. Else the sequences in the config file are used.

However, for generating the ref data set report, we will have to use this new column.